framework/base/Event.php

```diff
@@ -153,9 +198,10 @@ public static function offAll()
      */
     public static function hasHandlers($class, $name)
     {
-        if (empty(self::$_events[$name])) {
+        if (empty(self::$_events[$name]) && empty(self::$_eventWildcards)) {
```

Comment:
Conditions could be inverted. Checking single variable costs less then accessing array index.

Reply:
optimized